zero.go

```diff
@@ -12,18 +12,23 @@ func Zero[T any](_ *T) T {
 // IsZero checks if a given value is zero, either using
 // the IsZero() bool interface or reflection
 func IsZero(vi any) bool {
-	if vi == nil {
+	switch p := vi.(type) {
+	case nil:
+		// nil
 		return true
```

Comment:
This will need to be mentioned in the documentation as it goes against the principle of least surprise. Some people might want to know if something is `nil` because they might want to initialize it or something. But then again a bool is a bool so they might do something like:

```go
if IsInitialized(d) && IsZero(d) {
...
}
```

Reply:
it doesn't really change the behaviour as otherwise `nil` would be caught by reflect's `IsValid()`
personally I use IsZero() to determine if I should initialise something to the default value, or if I should skip some bit of logic